src/mbgl/programs/program.hpp

```diff
@@ -1,33 +1,38 @@
 #pragma once
 
 #include <mbgl/gl/program.hpp>
+#include <mbgl/programs/program_parameters.hpp>
 
 #include <cassert>
 
 namespace mbgl {
 
-enum class ProgramDefines : bool {
-    None = false,
-    Overdraw = true,
-};
-
 template <class Shaders, class Primitive, class Attributes, class Uniforms>
 class Program : public gl::Program<Primitive, Attributes, Uniforms> {
 public:
     using ParentType = gl::Program<Primitive, Attributes, Uniforms>;
 
-    Program(gl::Context& context, ProgramDefines defines)
-        : ParentType(context, Shaders::vertexSource, fragmentSource(defines))
+    Program(gl::Context& context, const ProgramParameters& programParameters)
+        : ParentType(context, vertexSource(programParameters), fragmentSource(programParameters))
         {}
 
-    static std::string fragmentSource(ProgramDefines defines) {
+    static std::string fragmentSource(const ProgramParameters& programParameters) {
         std::string fragment = Shaders::fragmentSource;
-        if (defines == ProgramDefines::Overdraw) {
+        
+        fragment.replace(fragment.find_first_of('\n'), 1,"\n#define DEVICE_PIXEL_RATIO "+ std::to_string(programParameters.pixelRatio) + "\n");
```

Comment:
What's the purpose of finding the first newline here? Could it instead simply prepend `"#define DEVICE_PIXEL_RATIO ...\n"` to the source?